BUG: zipfian precision test failure on Windows / mingw-w64

### Describe your issue.

Tests on Scipy compiled with mingw-w64 give the following error:

```
_________________________________________ TestZipfian.test_zipfian_continuity _________________________________________
..\build\Lib\site-packages\scipy\stats\tests\test_discrete_distns.py:261: in test_zipfian_continuity
    assert_allclose(zipfian.sf(k, alt1, N), zipfian.sf(k, agt1, N))
E   AssertionError:
E   Not equal to tolerance rtol=1e-07, atol=0
E
E   Mismatched elements: 5 / 30 (16.7%)
E   Max absolute difference: 1.54294993e-08
E   Max relative difference: 3.85015542e-07
E    x: array([0.749686, 0.624529, 0.541092, 0.478513, 0.42845 , 0.386731,
E          0.350972, 0.319683, 0.291871, 0.266839, 0.244083, 0.223224,
E          0.203969, 0.186089, 0.169402, 0.153757, 0.139033, 0.125127,...
E    y: array([0.749686, 0.624529, 0.541092, 0.478513, 0.42845 , 0.386731,
E          0.350972, 0.319683, 0.29187 , 0.266839, 0.244083, 0.223224,
E          0.203969, 0.186089, 0.169402, 0.153757, 0.139033, 0.125127,...
        N          = 30
        agt1       = 1.00000001
        alt1       = 0.99999999
        k          = array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30])
```

The max and relative differences look pretty small.   Is it reasonable to relax the test tolerances here?

### Reproducing Code Example

```python
# Compile Scipy using mingw-w64 UCRT toolchain from:
# https://github.com/r-windows/rtools-ucrt
# Compile (using Meson)
import scipy.stats; scipy.stats.test()
```


### Error message

```shell
As above.
```


### SciPy/NumPy/Python version information

1.8.0.dev0+1822.a8d340e 1.21.3 sys.version_info(major=3, minor=9, micro=6, releaselevel='final', serial=0)

--- a/scipy/stats/tests/test_discrete_distns.py
+++ b/scipy/stats/tests/test_discrete_distns.py
@@ -256,11 +256,14 @@ class TestZipfian:
         alt1, agt1 = 0.99999999, 1.00000001
         N = 30
         k = np.arange(1, N + 1)
-        assert_allclose(zipfian.pmf(k, alt1, N), zipfian.pmf(k, agt1, N))
-        assert_allclose(zipfian.cdf(k, alt1, N), zipfian.cdf(k, agt1, N))
-        assert_allclose(zipfian.sf(k, alt1, N), zipfian.sf(k, agt1, N))
+        assert_allclose(zipfian.pmf(k, alt1, N), zipfian.pmf(k, agt1, N),
+                        rtol=5e-7)
+        assert_allclose(zipfian.cdf(k, alt1, N), zipfian.cdf(k, agt1, N),
+                        rtol=5e-7)
+        assert_allclose(zipfian.sf(k, alt1, N), zipfian.sf(k, agt1, N),
+                        rtol=5e-7)
         assert_allclose(zipfian.stats(alt1, N, moments='msvk'),
-                        zipfian.stats(agt1, N, moments='msvk'), rtol=2e-7)
+                        zipfian.stats(agt1, N, moments='msvk'), rtol=5e-7)
 
     def test_zipfian_R(self):
         # test against R VGAM package

